src/source/speed.rs

```diff
@@ -1,3 +1,50 @@
+//! Playback Speed control Module.
+//!
+//! The main concept of this module is the [`Speed`] struct, which
+//! encapsulates playback speed controls of the current sink.
+//!
+//! In order to speed up a sink, the speed struct:
+//! - Increases the current sample rate by the given factor
+//! - Updates the total duration function to cover for the new factor by dividing by the factor
+//! - Update the pos function by multiplying the position by the factor
+//!
+//! To speed up a source from sink all you need to do is call the   `set_speed(factor: f32)` function
+//! For example, here is how you speed up your sound by using sink or playing raw
+//!
+//! ```no_run
+//! use std::fs::File;
+//! use std::io::BufReader;
+//! use rodio::{Decoder, Sink, OutputStream, source::{Source, SineWave}};
+//!
+//! // Get an output stream handle to the default physical sound device.
+//! // Note that no sound will be played if _stream is dropped
+//! let (_stream, stream_handle) = OutputStream::try_default().unwrap();
+//! // Load a sound from a file, using a path relative to Cargo.toml
+//! let file = BufReader::new(File::open("examples/music.ogg").unwrap());
+//! // Decode that sound file into a source
+//! let source = Decoder::new(file).unwrap();
+//! // Play the sound directly on the device 2x faster
+//! stream_handle.play_raw(source.convert_samples().speed(2.0));
+//!
+//! // The sound plays in a separate audio thread,
+//! // so we need to keep the main thread alive while it's playing.
+//! std::thread::sleep(std::time::Duration::from_secs(5));
+//!
+//! // here is how you would do it using the sink
```

Comment:
should probably be two separate code "blocks" (I mean the ``` things). Also you can hide the imports using `#`.

Reply:
I did not quite get this though . Do you mean the examples

Reply:
yes the examples, I figured you put them in one code block to save having to show the setup code (imports etc) twice. However using # you can hide that.

Reply:
Argh, I get it now 

I would separate the examples so it is easier to understand 

And maybe the imports aren't that necessary ; I was just trying to conform to your docs pattern

Reply:
> And maybe the imports aren't that necessary ; I was just trying to conform to your docs pattern

rodio is pretty old so unfortunately a lot of existing documentation is from before imports could be hidden.

Reply:
Alright , I hid the documentation and separated the two examples